src/code/z_fbdemo.c

```diff
@@ -60,59 +75,67 @@ void TransitionUnk_InitGraphics(TransitionUnk* this) {
                 vtx2->cn[1] = 0;
                 vtx2->cn[2] = 120;
                 vtx2->cn[3] = 255;
+
                 rowTex += 0x20;
             }
+
+            colTex += 0x20;
         }
     }
 
     gfx = this->gfx;
-    for (colTex = 0, col = 0; col < this->col; colTex += 0x20, col++) {
-
-        gSPVertex(gfx++, SEGMENT_ADDR(0xA, (u32)col * (this->row + 1) * sizeof(Vtx)), 2 * (this->row + 1), 0);
-
-        for (rowTex = 0, row = 0, row2 = 0; row < this->row;) {
+    colTex = 0;
+    for (col = 0; col < this->cols; col++) {
+        gSPVertex(gfx++, SEGMENT_ADDR(0xA, (u32)col * (this->rows + 1) * sizeof(Vtx)), 2 * (this->rows + 1), 0);
+
+        rowTex = 0;
+        row2 = 0;
+        row = 0;
+        while (row < this->rows) {
             gDPPipeSync(gfx++);
 
             gDPLoadTextureTile(gfx++, SEGMENT_ADDR(0xB, 0), G_IM_FMT_RGBA, G_IM_SIZ_16b, SCREEN_WIDTH, SCREEN_HEIGHT,
                                rowTex, colTex, rowTex + 0x20, colTex + 0x20, 0, G_TX_NOMIRROR | G_TX_WRAP,
                                G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
 
-            gSP1Quadrangle(gfx++, row, row + 1, row2 + this->row + 2, this->row + row2 + 1, 0);
+            gSP1Quadrangle(gfx++, row, row + 1, row2 + this->rows + 2, this->rows + row2 + 1, 0);
 
             rowTex += 0x20;
             row2++;
             row++;
         }
+
+        colTex += 0x20;
     }
 
     gDPPipeSync(gfx++);
     gSPEndDisplayList(gfx++);
 
-    LOG_NUM("this->col * (1 + this->row * (1 + 7 + 1)) + 1 + 1", this->col * (1 + this->row * 9) + 2, "../z_fbdemo.c",
+    LOG_NUM("this->col * (1 + this->row * (1 + 7 + 1)) + 1 + 1", this->cols * (1 + this->rows * 9) + 2, "../z_fbdemo.c",
```

Comment:
not sure what the original "col" and "row" meant, but it's not "amount of columns" and "amount of rows"

unless they had vertical rows and horizontal columns

Reply:
Yeah, they do seem reversed. Lowering the numbers below 5 and 4 (not 100% sure why that is the limit since the screen seems divided by 10 and 7), I can control the number of rows and cols that get drawn. For instance with `cols=2` and `rows=3`.
![image](https://user-images.githubusercontent.com/31974442/209452373-6129fac6-4430-4e9e-9e34-6c27a84bc074.png)

Reply:
Consider using angrylion, a very accurate low level emulation video plugin, https://github.com/ata4/angrylion-rdp-plus , or the parallel video plugin (which idk how to get on pj64, it's typically bundled with other emulators though like ares or simple64),
instead of high level emulation video plugins, even if GlideN64 is very good for a HLE video plugin it will often struggle or act odd with weird stuff like this

> Lowering the numbers below 5 and 4 (not 100% sure why that is the limit since the screen seems divided by 10 and 7)

limit? what do you mean? (I didn't try lowering these myself)

Reply:
Okay, using simple64 it acts completely different (I should have known using PJ64 was going to mess with me someday). It more blends things to a point (if you use the dynamic version), where the cols and rows tell which part of the screen to blend. Static doesn't seem to do anything.

Reply:
With this information, perhaps `TransitionTile` isn't best, but maybe `TransitionBlend`.

Reply:
For reference this is with `cols=3` and `rows=5`
![image](https://user-images.githubusercontent.com/31974442/209453948-4fa10083-4f6c-4b20-a3c5-e93a709f343a.png)

Reply:
> Okay, using simple64 it acts completely different (I should have known using PJ64 was going to mess with me someday).

:sweat_smile: 

> It more blends things to a point (if you use the dynamic version), where the cols and rows tell which part of the screen to blend.

yeah

> Static doesn't seem to do anything.

visually nothing happens yeah, it just draws the tiles "where they belong" so there's nothing to see. 

> With this information, perhaps `TransitionTile` isn't best, but maybe `TransitionBlend`?

TransitionTile works for me since the main feature is splitting the saved framebuffer into "rectangles" (so "tiles") that can then be drawn at arbitrary locations

---------

Try the following patch (bottom of this message), I think it should help a lot clearing up what's happening here

(to apply a patch, run `git apply`, copypaste to console, then hit Ctrl+D, or save the patch to a file and `git apply path/to/patch.txt`)

This adds the following controls on controller 1:
- L to turn on/off TransitionTile
- while TransitionTile is on:
  - Z to clear the framebuffer using `sTransTileBackgroundDL` (as long as Z is held)
  - dpad to reset TransitionTile with different cols/rows values (e.g. dpad left decreases the amount of cols, "shrinking" horizontally)

The effect achieved is drawing the saved framebuffer, scaled down and moving across the screen.

Testing suggestion:

1. Load file 1 (for some reason using this on title screen is jank)
2. Press L
   Screenshot after half a second:
   <details>

   ![image](https://user-images.githubusercontent.com/5306619/209562143-b5c5c78d-3dfa-4ce8-924f-60a18974f0af.png)

   </details>
3. Observe the "trail" left by the moving image. This is because the framebuffer is not cleared
4. Hold Z. This clears the framebuffer every frame with `sTransTileBackgroundDL`
   Screenshot:
   <details>

   ![image](https://user-images.githubusercontent.com/5306619/209562580-c6709a48-bd8c-484c-978c-18fbedd7ed07.png)

   </details>
6. Let go of Z, if you want to
7. Press dpad down. you see garbage drawn, this is data past the valid framebuffer data saved in the zbuffer (note dpad right has code to prevent having more than 10 cols, since for a reason I am unaware of that crashes the game)
   Screenshot:
   <details>

   ![image](https://user-images.githubusercontent.com/5306619/209562707-71ca5609-827c-4f19-9632-b40008996d83.png)

   </details>
9. press dpad up/down/left/right to adjust the amount of cols/rows
   Screenshot:
   <details>

   ![image](https://user-images.githubusercontent.com/5306619/209562880-53e40e52-8fb5-4bd0-8ec6-61f9a32e065b.png)

   </details>

Note: this isn't crash-free and I didn't bother to debug why

<details>

<summary>

**>>> PATCH <<<**

</summary>

```diff
diff --git a/include/functions.h b/include/functions.h
index e59a2a51a..6cf9521f1 100644
--- a/include/functions.h
+++ b/include/functions.h
@@ -1210,7 +1210,7 @@ void func_800AD958(struct_80166500* this, Gfx** gfxp);
 void PlayerCall_InitFuncPtrs(void);
 void TransitionTile_Destroy(TransitionTile* this);
 TransitionTile* TransitionTile_Init(TransitionTile* this, s32 cols, s32 rows);
-void TransitionTile_Draw(TransitionTile* this, Gfx** gfxP);
+void TransitionTile_Draw(TransitionTile* this, Gfx** gfxP, PlayState* play);
 void TransitionTile_Update(TransitionTile* this);
 void TransitionTriforce_Start(void* thisx);
 void* TransitionTriforce_Init(void* thisx);
diff --git a/src/code/z_fbdemo.c b/src/code/z_fbdemo.c
index cf6f6a988..a1fcb67bc 100644
--- a/src/code/z_fbdemo.c
+++ b/src/code/z_fbdemo.c
@@ -204,10 +204,12 @@ void TransitionTile_SetVtx(TransitionTile* this) {
     }
 }
 
-void TransitionTile_Draw(TransitionTile* this, Gfx** gfxP) {
+void TransitionTile_Draw(TransitionTile* this, Gfx** gfxP, PlayState* play) {
     Gfx* gfx = *gfxP;
 
-    gSPDisplayList(gfx++, sTransTileSetupDL);
+    if (play->state.input[0].cur.button & BTN_Z) {
+        gSPDisplayList(gfx++, sTransTileBackgroundDL);
+    }
     TransitionTile_SetVtx(this);
     gSPMatrix(gfx++, &this->projection, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_PROJECTION);
     gSPMatrix(gfx++, &this->modelView, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
@@ -246,7 +248,36 @@ void TransitionTile_Suck(TransitionTile* this) {
     }
 }
 
+f32 fmodf(f32 x, f32 y);
+
+void TransitionTile_Test(TransitionTile* this) {
+    s32 row;
+    s32 col;
+    f32 diffX;
+    f32 diffY;
+    f32 scale;
+    static s32 timer = 0;
+    f32 tileWidth = 20;
+    f32 tileHeight = 20;
+    f32 dx, dy;
+
+    timer++;
+    dx = fmodf(timer, SCREEN_WIDTH - this->cols * tileWidth);
+    dy = fmodf(timer, SCREEN_HEIGHT - this->rows * tileHeight);
+
+    for (row = 0; row < this->rows + 1; row++) {
+        for (col = 0; col < this->cols + 1; col++) {
+            f32 x, y;
+            x = col * tileWidth + dx;
+            y = row * tileHeight + dy;
+            (this->vtxData + col + row * (this->cols + 1))->x = CLAMP(x, 0, SCREEN_WIDTH);
+            (this->vtxData + col + row * (this->cols + 1))->y = CLAMP(y, 0, SCREEN_HEIGHT);
+        }
+    }
+}
+
 void TransitionTile_Update(TransitionTile* this) {
+    TransitionTile_Test(this);
 }
 
 s32 func_800B23F0(TransitionTile* this) {
diff --git a/src/code/z_play.c b/src/code/z_play.c
index f2640886c..51daf445e 100644
--- a/src/code/z_play.c
+++ b/src/code/z_play.c
@@ -455,6 +455,15 @@ void Play_Update(PlayState* this) {
 
     input = this->state.input;
 
+    if (input->press.button & BTN_L) {
+        if (gTransitionTileState == TRANS_TILE_OFF) {
+            gTransitionTileState = TRANS_TILE_SETUP;
+        } else if (gTransitionTileState == TRANS_TILE_DONE) {
+            TransitionTile_Destroy(&sTransitionTile);
+            gTransitionTileState = TRANS_TILE_OFF;
+        }
+    }
+
     if ((SREG(1) < 0) || (DREG(0) != 0)) {
         SREG(1) = 0;
         ZeldaArena_Display();
@@ -493,9 +502,36 @@ void Play_Update(PlayState* this) {
         }
 
         if (gTransitionTileState != TRANS_TILE_OFF) {
+            static s32 cur_cols = 10, cur_rows = 7;
+            s32 cols = cur_cols, rows = cur_rows;
+            s32 refresh = false;
+            u16 pressBtn = this->state.input[0].press.button;
+
+            if (pressBtn & BTN_DLEFT) {
+                cols--;
+            }
+            if (pressBtn & BTN_DRIGHT) {
+                cols++;
+            }
+            if (pressBtn & BTN_DUP) {
+                rows--;
+            }
+            if (pressBtn & BTN_DDOWN) {
+                rows++;
+            }
+            cols = CLAMP_MIN(cols, 0);
+            cols = CLAMP_MAX(cols, SCREEN_WIDTH / 32);
+            rows = CLAMP_MIN(rows, 0);
+
+            refresh = cols != cur_cols || rows != cur_rows;
+            if (refresh) {
+                cur_cols = cols;
+                cur_rows = rows;
+            }
+
             switch (gTransitionTileState) {
                 case TRANS_TILE_PROCESS:
-                    if (TransitionTile_Init(&sTransitionTile, 10, 7) == NULL) {
+                    if (TransitionTile_Init(&sTransitionTile, cur_cols, cur_rows) == NULL) {
                         osSyncPrintf("fbdemo_init呼出し失敗！\n"); // "fbdemo_init call failed!"
                         gTransitionTileState = TRANS_TILE_OFF;
                     } else {
@@ -506,7 +542,12 @@ void Play_Update(PlayState* this) {
                     break;
 
                 case TRANS_TILE_DONE:
-                    TransitionTile_Update(&sTransitionTile);
+                    if (refresh) {
+                        TransitionTile_Destroy(&sTransitionTile);
+                        gTransitionTileState = TRANS_TILE_SETUP;
+                    } else {
+                        TransitionTile_Update(&sTransitionTile);
+                    }
                     break;
 
                 default:
@@ -1116,7 +1157,7 @@ void Play_Draw(PlayState* this) {
         if (gTransitionTileState == TRANS_TILE_DONE) {
             Gfx* sp88 = POLY_OPA_DISP;
 
-            TransitionTile_Draw(&sTransitionTile, &sp88);
+            TransitionTile_Draw(&sTransitionTile, &sp88, this);
             POLY_OPA_DISP = sp88;
             goto Play_Draw_DrawOverlayElements;
         } else {
```

</details>

Reply:
> TransitionTile works for me since the main feature is splitting the saved framebuffer into "rectangles" (so "tiles") that can then be drawn at arbitrary locations

Makes sense.